### ISSUE TITLE
Consider Switching Java back to 4 Space Indentation

### ISSUE THREAD

**Issue by InsertCreativityHere** [zeroc-ice/ice#2160]
We used to indent our Java code with 4 spaces, but when we added the spotless formatting plugin to our build, it switched to 2 spaces, since it's using `googleJavaFormat`, and Google's style guidelines dictate 2 spaces for block indentation. But this 2-space Java feels pretty unnatural to me. In my experience with Java, this style of indentation is pretty rare and 4 spaces is pretty standard.

But... to double check myself, I looked at the [Github Trending Java Projects](https://github.com/trending/java?since=daily).
The first half of that page are indeed all using 4 spaces (I didn't check the 2nd half).

- [Apache Doris](https://github.com/search?q=repo%3Aapache%2Fdoris++language%3AJava&type=code): 4 spaces
- [Kubernetes Kafka](https://github.com/search?q=repo%3Astrimzi%2Fstrimzi-kafka-operator++language%3AJava&type=code): 4 spaces
- [OpenAI Generator](https://github.com/search?q=repo%3AOpenAPITools%2Fopenapi-generator++language%3AJava&type=code): 4 spaces
- [Apache Dolphin Scheduler](https://github.com/search?q=repo%3Aapache%2Fdolphinscheduler++language%3AJava&type=code): 4 spaces
- [Stirling PDF](https://github.com/search?q=repo%3AStirling-Tools%2FStirling-PDF++language%3AJava&type=code): 4 spaces
- [Skylot Jadx](https://github.com/search?q=repo%3Askylot%2Fjadx++language%3AJava&type=code): 4 spaces
- [Keycloak](https://github.com/search?q=repo%3Akeycloak%2Fkeycloak++language%3AJava&type=code): 4 spaces

Feel free to check around yourselves. These aren't cherry-picked or anything,
they're literally just the top 7 results for [Github Trending Java Projects](https://github.com/trending/java?since=daily) (ignoring the Chinese ones I couldn't read).

Also, for good measure, I checked some other big player's Java repositories just to see:

- [Microsoft Java-Debug](https://github.com/search?q=repo%3Amicrosoft%2Fjava-debug++language%3AJava&type=code): 4 spaces
- [Microsoft Malmo AI](https://github.com/search?q=repo%3Amicrosoft%2Fmalmo++language%3AJava&type=code): 4 spaces
- [MS-SQL Java](https://github.com/search?q=repo%3Amicrosoft%2Fmssql-jdbc++language%3AJava&type=code): 4 spaces
_

- [Apple FoundationDB](https://github.com/search?q=repo%3Aapple%2Ffoundationdb++language%3AJava&type=code): 8 spaces
- [Apple PKL-core](https://github.com/search?q=repo%3Aapple%2Fpkl++language%3AJava&type=code): 2 spaces
- [Apple XCode JDK](https://github.com/search?q=repo%3Aapple%2Fopenjdk++language%3AJava&type=code): 4 spaces
_

- [AWS Java SDK](https://github.com/search?q=repo%3Aaws%2Faws-sdk-java-v2++language%3AJava&type=code): 4 spaces
- [AWS Lambda Java](https://github.com/search?q=repo%3Aaws%2Fserverless-java-container++language%3AJava&type=code): 4 spaces
- [Amazon ExoPlayer](https://github.com/search?q=repo%3Aamzn%2Fexoplayer-amazon-port++language%3AJava&type=code): 2 spaces
- [Amazon Pay Java](https://github.com/search?q=repo%3Aamzn%2Famazon-pay-sdk-java++language%3AJava&type=code): 4 spaces
_

- [Oracle Tribuo](https://github.com/search?q=repo%3Aoracle%2Ftribuo++language%3AJava&type=code): 4 spaces
- [Oracle Graal](https://github.com/search?q=repo%3Aoracle%2Fgraal++language%3AJava&type=code): 4 spaces
- [Oracle VisualVM](https://github.com/search?q=repo%3Aoracle%2Fvisualvm++language%3AJava&type=code): 4 spaces
- [Oracle FastR](https://github.com/search?q=repo%3Aoracle%2Ffastr++language%3AJava&type=code): 4 spaces
_

- [OpenJDK JDK](https://github.com/search?q=repo%3Aopenjdk%2Fjdk++language%3AJava&type=code): 4 spaces
- [OpenJDK Mobile](https://github.com/search?q=repo%3Aopenjdk%2Fmobile++language%3AJava&type=code): 4 spaces
- [OpenJDK Mission Control](https://github.com/search?q=repo%3Aopenjdk%2Fjmc++language%3AJava&type=code): 8 spaces

OpenJDK, Oracle, Amazon, and Microsoft seem to have clear 4 space standards (with 2 exceptions).
Apple doesn't seem to have any consistency, using a mix of 2, 4, and 8 spaces.
I didn't bother to check Google, because I assume they're using their own guideline of 2 spaces.

----

Note that Google's style guidelines are not Java specific.
The fact that they use 2 spaces in Java is because they also use 2 spaces in C++
https://google.github.io/styleguide/cppguide.html#Spaces_vs._Tabs

**Comment by InsertCreativityHere** [comment]
Also, to disavow myself of my biases here:
IMO, 4 space indentation is objectively easier to read than 2 spaces
spoken as someone who is currently editing large chunks of the Java tests.

**Comment by pepone** [comment]
I'm fine with 4 or 2, I don't think it makes much of a difference. But I would prefer to keep the Google style default, which is the point of using a style, if we were to start tweaking things here and there we could keep the original style.

**Comment by InsertCreativityHere** [comment]
I agree that we shouldn't spend time tweaking and modifying the defaults.
My thinking was just switching which format-style we were using, but keeping it's defaults intact.

It looks like spotless supports a couple styles if you scroll to the "java" section of [this](https://github.com/diffplug/spotless?tab=readme-ov-file#current-feature-matrix):
- java.GoogleJavaFormatStep
- java.PalantirJavaFormatStep
- java.EclipseJdtFormatterStep

The [`PalantirJavaFormat`](https://github.com/palantir/palantir-java-format) style seems like it was specifically made to "fix" the google style, after reading it's README. Maybe they're biased here though.

----

Also, it looks like spotless has support for other lints apart from just formatting:
- java.RemoveUnusedImportsStep
- java.ImportOrderStep
- java.FormatAnnotationsStep

Should we try enabling any of these? If they work well, they seem like good checks to be performing.

**Comment by externl** [comment]
It would be nice to be able to use the very common [Language Support for Java(TM) by Red Hat](https://marketplace.visualstudio.com/items?itemName=redhat.java) Java extension for formatting. It seems to use the eclipse formatter.

https://github.com/redhat-developer/vscode-java/wiki/Formatter-settings

**Comment by bernardnormier** [comment]
Spotless itself uses tabs for indentation with indent-size = 4.
https://github.com/diffplug/spotless/blob/04d5f28c0401803323b3e8d82677fc081ab2b818/.editorconfig#L5

(I don't think we should use tabs)

**Comment by InsertCreativityHere** [comment]
This was implemented in #2734.
